bloom-filter/src/main/scala/bloomfilter/mutable/CuckooFilter.scala

```diff
@@ -64,7 +65,7 @@ object CuckooFilter {
   }
 
   def optimalNumberOfBuckets(numberOfItems: Long): Long = {
-    var numberOfBuckets = upperPowerOf2(numberOfItems / UnsafeTable16Bit.TagsPerBucket)
+    var numberOfBuckets = upperPowerOf2((numberOfItems + UnsafeTable16Bit.TagsPerBucket - 1) / UnsafeTable16Bit.TagsPerBucket)
```

Comment:
Can it be more explicit? like
 `var numberOfBuckets = upperPowerOf2(math.max(numberOfItems / UnsafeTable16Bit.TagsPerBucket, 1L))` or even `if numberOfItems < TagsPerBucket ...`
Otherwise it could affect the number of buckets if numberOfItems is close to power of 2 then "+ UnsafeTable16Bit.TagsPerBucket - 1" could add it to the next power of 2. The fraction check helps here but still. WDYT?